main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintGen.scala

```diff
@@ -158,6 +158,19 @@ object ConstraintGen {
         val resEff = evar
         (resTpe, resEff)
 
+      case Expr.ApplyLocalDef(sym, exps, itvar, tvar, evar, loc) =>
+        val (tpes, effs) = exps.map(visitExp).unzip
+        val defResTpe = Type.freshVar(Kind.Star, loc.asSynthetic)
```

Comment:
since defResTpe is exactly the same as tvar, you don't have to make a new typevariable

delete `defResTpe` and use `tvar` instead where it was used

Reply:
but it looks correct